source/server/configuration_impl.cc

```diff
@@ -109,27 +110,42 @@ void MainImpl::initializeTracers(const Json::Object& configuration) {
   std::string type = driver->getString("type");
   log().info(fmt::format("  loading tracing driver: {}", type));
 
-  ASSERT(type == "lightstep");
   ::Runtime::RandomGenerator& rand = server_.random();
-  Json::ObjectPtr lightstep_config = driver->getObject("config");
 
-  std::unique_ptr<lightstep::TracerOptions> opts(new lightstep::TracerOptions());
-  opts->access_token =
-      server_.api().fileReadToEnd(lightstep_config->getString("access_token_file"));
-  StringUtil::rtrim(opts->access_token);
+  if (type == "lightstep") {
+    Json::ObjectPtr lightstep_config = driver->getObject("config");
 
-  if (server_.localInfo().clusterName().empty()) {
-    throw EnvoyException("cluster name must be defined if LightStep tracing is enabled. See "
-                         "--service-cluster option.");
-  }
-  opts->tracer_attributes["lightstep.component_name"] = server_.localInfo().clusterName();
-  opts->guid_generator = [&rand]() { return rand.random(); };
+    std::unique_ptr<lightstep::TracerOptions> opts(new lightstep::TracerOptions());
+    opts->access_token =
+        server_.api().fileReadToEnd(lightstep_config->getString("access_token_file"));
+    StringUtil::rtrim(opts->access_token);
+
+    if (server_.localInfo().clusterName().empty()) {
+      throw EnvoyException("cluster name must be defined if LightStep tracing is enabled. See "
+                           "--service-cluster option.");
+    }
+    opts->tracer_attributes["lightstep.component_name"] = server_.localInfo().clusterName();
+    opts->guid_generator = [&rand]() { return rand.random(); };
+
+    Tracing::DriverPtr lightstep_driver(
+        new Tracing::LightStepDriver(*lightstep_config, *cluster_manager_, server_.stats(),
+                                     server_.threadLocal(), server_.runtime(), std::move(opts)));
+    http_tracer_.reset(
+        new Tracing::HttpTracerImpl(std::move(lightstep_driver), server_.localInfo()));
+  } else if (type == "zipkin") {
+    if (server_.localInfo().clusterName().empty()) {
```

Comment:
nit: remove else statement and add ASSERT checking for type is zipkin.  ex https://github.com/lyft/envoy/blob/master/source/common/router/router_ratelimit.cc#L102